### ISSUE TITLE
overwriting cofing by --options does NOT work 

### ISSUE THREAD

**Issue by maxiuw** [Pointcept/Pointcept#380]
I realized that when running training and adding specific `--options` to be overwritten in the confing does not necessarly work. 

When it works: 
e.g. we have `configs/nuscenes/semseg-spunet-v1m1-0-base.py`
```
# misc custom setting
batch_size = 12  # bs: total bs in all gpus
mix_prob = 0
empty_cache = False
enable_amp = True
```
if these variables are not later used in the other dict like model, optimizer ect it works so we could run e.g.
`python tools/train.py configs/nuscenes/semseg-spunet-v1m1-0-base.py --options batch_size=2`

But if e.g. we would define `learning_rate` at the top and then use it in optimizer i.e.
 we would add to config
`learning_rate = 0.002`
and then later couple lines below we woud use 
`optimizer = dict(type="AdamW", lr=learning_rate, weight_decay=0.005) # default 0.002`

it will still be build with the default learning rate so
`python tools/train.py configs/nuscenes/semseg-spunet-v1m1-0-base.py --options learning_rate=0.01` 
does not really takes effect. Doing --options optimizer.lr=0.01` also does not seem to be working (if i remember correctly in MMDET3D you would need to be explicit like that). 

What is the solution for that ? I was trying to understand how you read the config ect but i couldn't wrap my head around how to solve that issue...
